### PR TITLE
DM-18666 Add support for authorization headers to firefly_client

### PR DESCRIPTION
**Pull request by stargaser**
Add support for a string token to be passed to firefly_client.

To make this work, a `token` argument has been added to the `__init__` and factory methods.

### CODE REVIEW

File: firefly_client/firefly_client.py [hunk 1]
```diff
@@ -182,11 +188,12 @@ def make_lab_client(start_browser_tab=False, html_file=_my_html_file, start_tab=
                 print('     Safari: shows an animation to follow on left side bar')
         return FireflyClient(url=_my_url, html_file=html_file,
                              use_lab_env=True, start_tab=start_tab,
-                             start_browser_tab=start_browser_tab)
+                             start_browser_tab=start_browser_tab,
+                             token=token)
```
Comment (robyww) [inline firefly_client/firefly_client.py:196]: _from standup_: when you get a token and the protocol is http throw and exception.

**Review by robyww — APPROVED** [review]
Looks good. The parameter passing of the token makes sense.

**Review by gpdf — CHANGES_REQUESTED** [review]
I believe that we should not be sending the token in an authorization header if the connection is not secure, i.e., if it is an HTTP / "http://" session.

As the code is already checking for "http://" URLs, I think it would be straightforward for it to throw an exception if that branch is taken when a token has been supplied to the constructor.

File: firefly_client/firefly_client.py (hunk 1 above)
Comment (stargaser) [reply firefly_client/firefly_client.py:196]: Added to the `__init__` method.

File: firefly_client/firefly_client.py [hunk 2]
```diff
@@ -63,6 +63,9 @@ class FireflyClient(WebSocketClient):
         If True, bring up a Jupyterlab or a browser tab for Firefly. Default False.
     start_browser_tab: `bool`
         If True, and start_tab is true and in use_lab_env is true then start a browser tab. Default False.
+    token: `str` or None
+        A token for connecting to a Firefly server that requires
+        authentication.
```
Comment (gpdf) [inline firefly_client/firefly_client.py]: Can you state here in the docstrings what HTTPS header is actually set when a token is provided?

Comment (gpdf) [reply firefly_client/firefly_client.py]: (I think it's going to be "Authorization: Bearer" but I think this should just be stated explicitly to aid future users.)

**Review by gpdf — CHANGES_REQUESTED** [review]
In addition to the few words of docstring documentation about the "Authorization:" header, please also add a couple of sentences about the use of a token to the "user guide" documentation that appears at firefly-client.lsst.io.

File: firefly_client/firefly_client.py (hunk 2 above)
Comment (stargaser) [reply firefly_client/firefly_client.py]: I'll add this but I worry this strays from "how to use it" into "how it works".

**Comment by stargaser** [comment]
@gpdf - while it would be good to get into the habit of documenting new features in the pull requests that add them, I would prefer to document the authorization token as part of #39 (DM-16682). That PR includes a major reorganization of the documentation including showing the three different ways to initialize a `FireflyClient` instance. I can tack on an "Authorization" section in this PR, but it will have to be re-done in #39 anyway.

**Pull request merged**